81.md

```diff
@@ -0,0 +1,38 @@
+NIP-81
+======
+
+Private Relationship Status
+---------------------------
+
+`draft` `optional`
+
+A new `kind:30382` event documents a "Private Relationship Status" between two users of the network. 
+
+It is defined as a _parameterized replaceable event_ with a single `d` tag as a random UUID per pubkey. An optional `n` tag adds the target key to an unbound list.
+
+The other tags are stringified, NIP-44-encrypted and placed inside the `.content` of the event. 
+
+For example:
+
+```js
+{
+  "kind": 30382,
+  "tags": [
+    ["d", "<randomUUID>"],
```

Comment:
I was trying to hash as well but I wasn't completely sure it was safe given that a similar process is used to determine the private conversation key between two pubkeys using NIP44

Reply:
I think hmac with sha256 and nostr privkeys have the same number of possible combinations so it would be just as hard to crack it, so using hmac wouldn't be lowering the security of the privkey. Though I may be wrong @paulmillr @mikedilger @Semisol to the rescue?

Reply:
yeah this is a bad idea. 1. hkdf should be used for these cases instead of hmac. 2. keys should not be reused between protocols (MAC vs elliptic curve key).

Reply:
@paulmillr got confused with the 2nd statement. Like can we safely use hkdf with nostr privkey to deterministically hash a text? If yes, can you please give us an example of how would it be with your '@noble/hashes/hkdf' lib?

Reply:
why does it need to contain user's private key and not public?

Reply:
> why does it need to contain user's private key and not public?

Too easy to find out the hash of two pubkeys used in Nostr. You can just have a big table of hashes for every 2 USED pubkeys and know who is talking to who (which should be private).

Reply:
The main idea is to hide from public a string value in a way no one without something only I have (my privkey) can guess the original value by hashing a list of strings the person thinks may include the original string.

I mean, if I have a list of people I follow (pubkeys), if I hash a pubkey simply with `sha256(string)`, people will easily guess which one by doing sha256 on all my follow list.

Reply:
I suggest this:

```
nip81_key = hkdf(private_key, salt: 'nip81') 
bobs_tag = sha256(nip81_key || bobs_pub)
kates_tag = sha256(nip81_key || kates_pub)
```

Reply:
the less stuff we do with private keys, the less chance it would accidentally leak, etc. nip81 key can be cached and passed anywhere, since it's less important